### ISSUE TITLE
Add -I and -D flags to occa {translate,compile} options

### ISSUE THREAD

**Issue by dmed256** [libocca/occa#153]
Mentioned in #150, add some commonly used compiler flags to `occa translate` and `occa compile`

**Comment by dmed256** [comment]
Added in 7055d922

**Comment by dmed256** [comment]
@pdhahn Added `-D` and `-I` options with a small bug/lack-of-feature that should be fixed in #154 
### Wrong
```bash
-Dfoo
-Dfoo=3
```
### Correct
```bash
-D foo
-D foo=3
```
### CLI 
```bash
> occa translate --help

Usage: occa translate [OPTIONS] FILE

Translate kernels

Arguments:
  FILE    An .okl file

Options:
  -D, --define          Add additional define
  -I, --include-path    Add additional include path
  -k, --kernel-props    Kernel properties
  -m, --mode            Output mode
```
```bash
> occa compile --help

Usage: occa compile [OPTIONS] FILE KERNEL

Compile kernels

Arguments:
  FILE      An .okl file
  KERNEL    Kernel name

Options:
  -D, --define          Add additional define
  -I, --include-path    Add additional include path
  -d, --device-props    Device properties
  -k, --kernel-props    Kernel properties
```
